corehq/blobs/management/commands/export_selected_blobs.py

```diff
@@ -32,31 +36,79 @@ class Command(BaseCommand):
+                            help='Only export blobs for metadata in files with filenames matching the filter regex.')
         parser.add_argument('--already_exported', dest='already_exported',
                             help='Pass a file with a list of blob names already exported')
+        parser.add_argument('--use-extracted', action='store_true', default=False, dest='use_extracted',
+                            help="Use already extracted dump if it exists.")
+        parser.add_argument('--output-path', help='Directory to write data files to.')
         parser.add_argument('--json-output', action="store_true", help="Produce JSON output for use in tests")
 
     @change_log_level('boto3', logging.WARNING)
     @change_log_level('botocore', logging.WARNING)
     def handle(self, path, **options):
+        use_extracted = options.get('use_extracted')
+        output_path = options.get('output_path') or ''
+
         already_exported = get_lines_from_file(options['already_exported'])
-        print("Found {} existing blobs, these will be skipped".format(len(already_exported)))
+        if already_exported:
+            print("Found {} existing blobs, these will be skipped".format(len(already_exported)))
+
+        if output_path and (not os.path.exists(output_path) or os.path.isfile(output_path)):
+            raise CommandError("Output path must exist and be a folder.")
+
+        filter_pattern = options.get('meta-file-filter')
+        filter_rx = None
+        if filter_pattern:
+            filter_rx = re.compile(filter_pattern)
+
+        def _filter(filename):
+            return 'blob_meta' in filename and (not filter_rx or filter_rx.match(filename))
+
+        target_dir = get_tmp_extract_dir(path, specifier='blob_meta')
+        target_path = Path(target_dir)
+        export_meta_files = []
+        if not target_path.exists():
+            with zipfile.ZipFile(path, 'r') as archive:
+                archive.extract('meta.json', target_dir)
+                for dump_file in archive.namelist():
+                    if 'blob_meta' in dump_file:
+                        archive.extract(dump_file, target_dir)
+        elif not use_extracted:
+            raise CommandError(
+                "Extracted dump already exists at {}. Delete it or use --use-extracted".format(target_dir))
+
+        meta = json.loads(target_path.joinpath('meta.json').read_text())
+        for file in target_path.iterdir():
+            if _filter(file.name):
+                export_meta_files.append(file)
+
+        results = []
+        filenames = []
+        with futures.ThreadPoolExecutor(max_workers=len(export_meta_files)) as executor:
+            for path in export_meta_files:
+                results.append(executor.submit(_export_blobs, output_path, path, meta))
+
+            for result in futures.as_completed(results):
+                filenames.append(result.result())
+
+            executor.shutdown()
```

Comment:
Should this be in a `finally:` block?

Reply:
I don't think this is strictly necessary

Reply:
removed